pkg/controller/controller_binding.go

```diff
@@ -365,13 +365,13 @@ func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[st
 	}
 
 	found := false
-
 	_, err := c.kubeClient.Core().Secrets(binding.Namespace).Get(binding.Spec.SecretName, metav1.GetOptions{})
 	if err == nil {
 		found = true
 	}
 
 	if found {
+		// TODO: Check ownerReferences first?
```

Comment:
Yes, must check owner references if they are used (they should?) + make this update a conditional update (CAS) with retry loop on conflict using resource version field. Value for it should be obtained from the GET above ^